server/src/main/java/org/elasticsearch/index/analysis/Analysis.java

```diff
@@ -353,10 +354,27 @@ public static Reader getReaderFromFile(Environment env, String filePath, String
         }
     }
 
-    public static Reader getReaderFromIndex(String synonymsSet, SynonymsManagementAPIService synonymsManagementAPIService) {
+    public static Reader getReaderFromIndex(
+        String synonymsSet,
+        SynonymsManagementAPIService synonymsManagementAPIService,
+        boolean ignoreMissing
+    ) {
         final PlainActionFuture<PagedResult<SynonymRule>> synonymsLoadingFuture = new PlainActionFuture<>();
         synonymsManagementAPIService.getSynonymSetRules(synonymsSet, synonymsLoadingFuture);
-        PagedResult<SynonymRule> results = synonymsLoadingFuture.actionGet();
+
+        PagedResult<SynonymRule> results;
+        try {
+            results = synonymsLoadingFuture.actionGet();
+        } catch (ResourceNotFoundException e) {
+            if (ignoreMissing == false) {
+                throw e;
+            }
+            logger.warn(
```

Comment:
Should this be at the error level?

Reply:
I also wonder if we should handle all exceptions rather than just the resource not found one?
Maybe we should even retry if that's appropriate (based on the exception)? 
I am fine if we do that in a follow up but a comment here would help to understand why we do that.

Reply:
> Should this be at the error level?

I was discussing this with @mayya-sharipova - I think this is not something that prevents ES from working, it's more of a user error. I think WARN is the appropriate level.

> I also wonder if we should handle all exceptions rather than just the resource not found one?
That's a good point - Adding that as a log with ERROR level in ecb2f26edea41362fd836632188e107e6eac7cfa

> Maybe we should even retry if that's appropriate (based on the exception)?

I'm not sure, what errors would benefit from a retry?

Shard allocation already is tried a number of times, so I think we should be covered here. In case it can be fixed after some additional action, `reroute` can be used to recover the shard.

Reply:
> Shard allocation already is tried a number of times, so I think we should be covered here. In case it can be fixed after some additional action, reroute can be used to recover the shard.

Yep that's fine, let's monitor the exception we catch here to be proactive.

Regarding the catching of exceptions, I am not sure that only `ElasticsearchException` are thrown by the call to `actionGet`, should we catch all `Exception` instead?